File: Module.Persist.TPM/Model/History/HistoricalPromoStatus.cs
using Core.History;
using Module.Persist.TPM.Model.TPM;
using System;
namespace Module.Persist.TPM.Model.History {
    [AssociatedWith(typeof(PromoStatus))]
    public class HistoricalPromoStatus : BaseHistoricalEntity<System.Guid> {
        public string Name { get; set; }

        public string SystemName { get; set; }

        public string Color { get; set; }

    }
}

File: Module.Frontend.TPM/Client/model/tpm/promostatus/HistoricalPromoStatus.js
Ext.define('App.model.tpm.promostatus.HistoricalPromoStatus', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'PromoStatus',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true, timeZone: +3, convert: dateConvertTimeZone},
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'Name', type: 'string', hidden: false, isDefault: true },
        { name: 'SystemName', type: 'string', hidden: false, isDefault: true },
        { name: 'Color', type: 'string', hidden: false, isDefault: true }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalPromoStatuss',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});
